dbt_platform_helper/providers/ecs.py

```diff
@@ -73,18 +130,32 @@ def ecs_exec_is_available(self, cluster_arn: str, task_arns: List[str]):
 
             task_details = self.ecs_client.describe_tasks(cluster=cluster_arn, tasks=task_arns)
 
-            managed_agents = task_details["tasks"][0]["containers"][0]["managedAgents"]
+            container_details = task_details["tasks"][0]["containers"][0]
+            if container_details.get("managedAgents", None):
+                managed_agents = container_details["managedAgents"]
+            else:
+                raise PlatformException("No managed agent on ecs task")
             execute_command_agent_status = [
                 agent["lastStatus"]
                 for agent in managed_agents
                 if agent["name"] == "ExecuteCommandAgent"
             ][0]
             if execute_command_agent_status != "RUNNING":
-                time.sleep(1)
+                time.sleep(3)
 
         if execute_command_agent_status != "RUNNING":
             raise ECSAgentNotRunningException
 
+    def wait_for_task_to_register(self, cluster_arn: str, task_family: str, max_attempts: int = 20):
+        for attempt in range(max_attempts):
+            task_arns = self.get_ecs_task_arns(cluster_arn, task_family)
+            if task_arns:
+                return task_arns
+            time.sleep(3)
```

Comment:
SECONDS_BEFORE_RETRY?

Reply:
Since it's being used in multiple places, perhaps pulling out the retry functionality into a reusable function may be better at this point.

Reply:
I think we have maybe reached that point!